### PR TITLE
Raise exception if any ee tasks fail

### PR DESCRIPTION
**Pull request by gridcell**

### CODE REVIEW

File: task_base.py [hunk 1]
```diff
@@ -464,14 +475,25 @@ def wait(self):
                 sleeptime = max_sleep
             time.sleep(sleeptime)
 
+        if bool(self._failed_ee_tasks) is True:
+            num_failed_tasks = len(list(self._failed_ee_tasks.keys()))
+            raise EETaskError(ee_messages=self._failed_ee_tasks)
```
Comment (saanobhaai) [inline task_base.py]: In thinking about this, raising an exception means that the task will exit, right, which means we have no way of monitoring the status of any remaining ee tasks other than manually? Should we instead wait until all ee tasks are finished, and then report # failed, # succeeded, # cancelled, etc?

File: task_base.py [hunk 2]
```diff
@@ -464,14 +475,25 @@ def wait(self):
                 sleeptime = max_sleep
             time.sleep(sleeptime)
 
+        if bool(self._failed_ee_tasks) is True:
+            num_failed_tasks = len(list(self._failed_ee_tasks.keys()))
```
Comment (saanobhaai) [inline task_base.py]: Can remove this line

File: task_base.py [hunk 3]
```diff
@@ -8,6 +8,14 @@
 
 PROJECTS = "projects"
 
+class EETaskError(Exception):
+    def __init__(self, ee_statuses, *args, **kwargs):
```
Comment (saanobhaai) [inline task_base.py:12]: Shouldn't the arg be `ee_messages`?

**Review by saanobhaai — CHANGES_REQUESTED** [review]

File: task_base.py (hunk 1 above)
Comment (gridcell) [reply task_base.py]: this doesn't raise an exception until after the while loop, so all the queued EE tasks should be done

File: task_base.py (hunk 3 above)
Comment (gridcell) [reply task_base.py:12]: I named it that because of it's the result of `getTaskStatus`

**Pull request merged**